More type-hints added

Quix0r opened the pull request:
After a long period of testing and fixing wrong type-hints some earlier commits had introduced on my node, I can finally assume that these type-hint changes are safe enough for develop code. Yes, it isn't less and it was a bumpy ride. :-)

Quix0r committed:
Added more type-hints

Quix0r committed:
Added more type-hints

Quix0r committed:
Added more type-hints and documented a few methods

Quix0r committed:
Added more type-hints

Quix0r committed:
Added more type-hints for "App" classes

Quix0r committed:
Continued:
- more type-hints
- fixed incompatible method declarations

Quix0r committed:
Another incompatible method declaration fixed + type-hints added

Quix0r committed:
Ops, bad type-hint here

Quix0r committed:
MrPetovan brought the right one up:
> Since this is depending on remote systems, the log can quickly fill with unactionable messages.

Quix0r committed:
Fixed "Argument 4 passed to Friendica\Protocol\DFRN::processVerbs() must be of the type bool"

Quix0r committed:
Continued:
- added more type-hints
- added some missing documentation

Quix0r committed:
Added again more type-hints

Quix0r committed:
Added more type-hints

Quix0r committed:
Some fixes:
- $gsid's default value cannot sadly be 0, it now must be null to allow some
  code work
- added some more type-hints
- documented a bit more

Quix0r committed:
Fixed:
- $object_data['actor'] can be null, but Receiver::getReceivers()'s 2nd
  parameter expect it to be string

Quix0r committed:
Fix for non-existing record system.mobile_theme in config table

Quix0r committed:
Continued:
- added more type-hints
- added some documentation
- Contact::getAccountType() should only process string, not null

Quix0r committed:
Continued:
- added type-hints
- removed out-dated documentation
- added some missing documentation

Quix0r committed:
Ops, wrong type-hint here, must be string ($nickname can never be an integer).

Quix0r committed:
Worker::getWaitingJobForPID() can also return FALSE on failure ... :-(

Quix0r committed:
Also need to declare $profile or otherwise an invocation of
Receiver::getReceiverForActor() will fail.

Quix0r committed:
Continued:
- added more type-hints
- some methods in Diaspora returned void but integer was documented so I
  changed it to -1 to have a proper type-hint

Quix0r committed:
Fix for "Uncaught Exception TypeError: "Argument 1 passed to
Friendica\Model\APContact::unarchiveInbox() must be of the type string, null
given" error message

Quix0r committed:
Possible fix for

Uncaught Exception TypeError: "Return value of
Friendica\Model\APContact::getByURL() must be of the type array, bool returned"

Quix0r committed:
Nore fixes

Quix0r committed:
Some more type-hints added

Quix0r committed:
Continued:
- changed back to 'return false;' as other methods heavily rely on false instead
  of an empty array as pointed out by @heluecht@pirati.ca
- $fetched_contact should be initialized as an empty array, let's not make this
  code more crazier than it already is (see APContact::getByURL())

Quix0r committed:
Ops, wrong type

Quix0r committed:
Ops, wrong type again

Quix0r commented:
First error noticed:
````
PHP Fatal error: Declaration of Friendica\Test\Util\AppDouble::isLogged() must be compatible with Friendica\App::isLogged(): bool in /woodpecker/src/github.com/friendica/friendica/tests/Util/AppDouble.php on line 46 
````

Quix0r committed:
Fixed incompatible types

MrPetovan commented on src/Content/Widget.php:
```diff
@@ -45,7 +45,7 @@ class Widget
 	 * @throws \Friendica\Network\HTTPException\InternalServerErrorException
 	 */
-	public static function follow($value = "")
+	public static function follow(string $value = ""): string
```
While you're at it, please normalize quotes in the lines you edited.
```suggestion
	public static function follow(string $value = ''): string
```

MrPetovan commented on src/Module/Directory.php:
```diff
@@ -166,7 +166,7 @@ public static function formatEntry(array $contact, $photo_size = 'photo')
 			'name'         => $contact['name'],
 			'details'      => $details,
-			'account_type' => Model\Contact::getAccountType($contact['contact-type']),
+			'account_type' => (!empty($contact['contact-type']) ? Model\Contact::getAccountType($contact['contact-type']) : ''),
```
```suggestion
			'account_type' => Model\Contact::getAccountType($contact['contact-type'] ?? ''),
```

MrPetovan commented on src/Protocol/DFRN.php:
```diff
@@ -346,9 +346,9 @@ public static function relocate($owner, $uid)
+	 * @todo  Find proper type-hint for returned type
 	 */
-	private static function addHeader(DOMDocument $doc, $owner, $authorelement, $alternatelink = "", $public = false)
+	private static function addHeader(DOMDocument $doc, array $owner, string $authorelement, string $alternatelink = "", bool $public = false)
```
```suggestion
	private static function addHeader(DOMDocument $doc, array $owner, string $authorelement, string $alternatelink = '', bool $public = false)
```

MrPetovan commented on src/Protocol/DFRN.php:
```diff
@@ -1060,7 +1063,7 @@ public static function transmit($owner, $contact, $atom, $public_batch = false)
 	 * @todo  Find good type-hints for all parameter
 	 */
-	private static function fetchauthor(\DOMXPath $xpath, \DOMNode $context, $importer, $element, $onlyfetch, $xml = "")
+	private static function fetchauthor(\DOMXPath $xpath, \DOMNode $context, array $importer, string $element, bool $onlyfetch, string $xml = ""): array
```
```suggestion
	private static function fetchauthor(\DOMXPath $xpath, \DOMNode $context, array $importer, string $element, bool $onlyfetch, string $xml = ''): array
```

MrPetovan commented on src/Protocol/Diaspora.php:
```diff
@@ -321,14 +321,14 @@ public static function decodeRaw(string $raw, string $privKey = '', bool $no_exi
 		if (!is_object($basedom)) {
 			Logger::notice('XML is not parseable.');
-			return false;
+			return [];
```
Please update the documentation to include the possibility of empty array in return values.

Quix0r replied:
Some outside methods rely on returned `false` in case of error. I guess this may need some rewrite later to have returned type-hint `array` back.

MrPetovan commented on src/Protocol/Diaspora.php:
```diff
@@ -748,15 +748,15 @@ private static function validPosting($msg)
-			return $r["pubkey"];
+		$fcontact = FContact::getByURL($handle);
+		if ($fcontact) {
+			return $fcontact["pubkey"];
```
```suggestion
			return $fcontact['pubkey'];
```

MrPetovan commented on src/Protocol/Diaspora.php:
```diff
@@ -2145,8 +2164,10 @@ private static function receiveRequestMakeFriend(array $importer, array $contact
-	private static function receiveContactRequest(array $importer, $data)
+	private static function receiveContactRequest(array $importer, $data): bool
 	{
+		// @TODO Need to find object type, roland@f.haeder.net
+		Logger::debug('data='.get_class($data));
```
Do you still need this? `$data` probably is a `stdClass` object.

Quix0r replied:
If you are sure?

Quix0r replied:
I have closed the other mentions as they are about the same added logger message. And I haven't found it yet.

Quix0r replied:
I have found `SimpleXMLElement` as a possible type. `receiveAccountDeletion()` receives `$fields` from `dispatch()` method which has that type-hint set.

MrPetovan commented on src/Protocol/Diaspora.php:
```diff
@@ -2388,12 +2409,14 @@ private static function addReshareActivity($item, $parent_message_id, $guid, $au
-	private static function receiveReshare(array $importer, $data, $xml, int $direction)
+	private static function receiveReshare(array $importer, $data, string $xml, int $direction): bool
 	{
+		// @TODO Need to find object type, roland@f.haeder.net
+		Logger::debug('data='.get_class($data));
```
Do you still need this? `$data` probably is a `stdClass` object.

MrPetovan commented on src/Protocol/Diaspora.php:
```diff
@@ -2512,8 +2535,10 @@ private static function receiveReshare(array $importer, $data, $xml, int $direct
-	private static function itemRetraction(array $importer, array $contact, $data)
+	private static function itemRetraction(array $importer, array $contact, $data): bool
 	{
+		// @TODO Need to find object type, roland@f.haeder.net
+		Logger::debug('data='.get_class($data));
```
Do you still need this? `$data` probably is a `stdClass` object.

MrPetovan commented on src/Protocol/Diaspora.php:
```diff
@@ -2578,8 +2603,10 @@ private static function itemRetraction(array $importer, array $contact, $data)
-	private static function receiveRetraction(array $importer, $sender, $data)
+	private static function receiveRetraction(array $importer, string $sender, $data)
 	{
+		// @TODO Need to find object type, roland@f.haeder.net
+		Logger::debug('data='.get_class($data));
```
Do you still need this? `$data` probably is a `stdClass` object.

MrPetovan commented on src/Protocol/Diaspora.php:
```diff
@@ -2656,6 +2683,8 @@ private static function isSolicitedMessage(array $item, string $author, string $
 	 */
 	private static function storePhotoAsMedia(int $uriid, $photo)
 	{
+		// @TODO Need to find object type, roland@f.haeder.net
+		Logger::debug('photo='.get_class($photo));
```
Do you still need this? `$data` probably is a `stdClass` object.

MrPetovan commented on src/Protocol/Diaspora.php:
```diff
@@ -3103,18 +3132,18 @@ private static function buildAndTransmit(array $owner, array $contact, $type, $m
 		// Don't send notifications for private postings
 		if ($item['private'] == Item::PRIVATE) {
-			return;
+			return -1;
```
What are the possible values? Why not `0` on error?

MrPetovan commented on src/Protocol/Diaspora.php:
```diff
@@ -3540,12 +3569,12 @@ private static function constructLike(array $item, array $owner)
 		}
 
-		return(["author" => self::myHandle($owner),
+		return ["author" => self::myHandle($owner),
 				"guid" => $item["guid"],
 				"parent_guid" => $parent["guid"],
 				"parent_type" => $target_type,
 				"positive" => $positive,
```
Please re-indent this array and convert quotes.
```suggestion
		return [
			'author' => self::myHandle($owner),
			'guid' => $item['guid'],
			'parent_guid' => $parent['guid'],
			'parent_type' => $target_type,
			'positive' => $positive,
			'author_signature' => '',
		];
```

MrPetovan commented on src/Protocol/Diaspora.php:
```diff
@@ -3579,11 +3608,11 @@ private static function constructAttend(array $item, array $owner)
 		}
 
-		return(["author" => self::myHandle($owner),
+		return ["author" => self::myHandle($owner),
```
Please re-indent this array and replace quotes.
```suggestion
		return [
			'author' => self::myHandle($owner),
```

Quix0r committed:
Changes:
- added some type-hints
- replaced most double-quotes (only Diaspora.php, later more) with single
- added some documentation
- normalized indenting in Diaspora.php (I hope I got all?)

Quix0r committed:
Fixed indenting

Quix0r committed:
Some outside code relies on returned "false"

Quix0r committed:
Set type-hint for parameter $data to SimpleXMLElement as $fields in dispatch()
is the same and being handled over.

Quix0r committed:
Fixed TypeError: "Argument 1 passed to Friendica\Core\System::httpExit() must be
of the type string, null given, called in /var/www/.../src/Module/DFRN/Poll.php
on line 37"

Quix0r committed:
Ops, needs to be a variable:

Error: "Cannot pass parameter 2 by reference" at /var/www/.../src/Module/DFRN/Poll.php line 36

MrPetovan commented on src/Protocol/Diaspora.php:
```diff
@@ -511,77 +518,77 @@ public static function dispatch(array $importer, array $msg, SimpleXMLElement $f
 
 			default:
-				Logger::notice("Unknown message type ".$type);
+				Logger::notice("Unknown message type " . $type);
```
Since we're going the full conversion route, let's pick up stragglers.
```suggestion
				Logger::notice('Unknown message type ' . $type);
```

nupplaphil replied:
Even better:
```
DI::logger->notice('Unknown message type', 'type' => $type);
```

So we're marking the Dependency Injection directly with 'DI' and we use parameter so we can distinct between message and payload of an log entry. This is helpful when we use enterprise logging systems like Splunk/Elastic

Quix0r replied:
@nupplaphil A lot lines then needs rework. I would find it more proper to move this to another `rewrite/log-messages` branch/PR instead of doing it here.

The other question that came up here is, when it is a unknown (e.g. not yet handled) message-type, I would find a warning proper than just a notice as maybe some `case 'foo':` is missing here then. But that's only how I would do it.

MrPetovan replied:
@nupplaphil I agree with you, but it isn't in the scope of this PR that's already humongous.

MrPetovan commented on src/Protocol/Diaspora.php:
```diff
@@ -621,106 +628,106 @@ private static function validPosting(array $msg)
 		$orig_type = $type;
 
-		Logger::debug("Got message type ".$type.": ".$msg["message"]);
+		Logger::debug("Got message type " . $type . ": " . $msg['message']);
```
```suggestion
		Logger::debug('Got message type ' . $type . ': ' . $msg['message']);
```

Quix0r replied:
Same as above with other message, maybe use `DI::logger->foo()` instead.

Quix0r replied:
But since this is very unknown to me (I personally would go with a `sprintf('Some message with %s some %d variables.')` to have log messages as a complete string.

MrPetovan commented on src/Protocol/Diaspora.php:
```diff
@@ -752,14 +759,14 @@ private static function key(string $handle): string
 		$handle = strval($handle);
 
-		Logger::notice("Fetching diaspora key for: ".$handle);
+		Logger::notice("Fetching diaspora key for: " . $handle);
```
```suggestion
		Logger::notice('Fetching diaspora key for: ' . $handle);
```

Quix0r replied:
Same as above. I used @nupplaphil 's suggested way as this was the easiest for me (only one parameter at the end).

MrPetovan commented on src/Protocol/Diaspora.php:
```diff
@@ -883,11 +890,11 @@ private static function postAllow(array $importer, array $contact, bool $is_comm
+		$contact = self::contactByHandle($importer['uid'], $handle);
 		if (!$contact) {
-			Logger::notice("A Contact for handle ".$handle." and user ".$importer["uid"]." was not found");
+			Logger::notice("A Contact for handle " . $handle . " and user " . $importer['uid'] . " was not found");
```
```suggestion
			Logger::notice('A Contact for handle ' . $handle . ' and user ' . $importer['uid'] . ' was not found');
```

MrPetovan commented on src/Protocol/Diaspora.php:
```diff
@@ -897,7 +904,7 @@ private static function allowedContactByHandle(array $importer, string $handle,
 
 		if (!self::postAllow($importer, $contact, $is_comment)) {
-			Logger::notice("The handle: ".$handle." is not allowed to post to user ".$importer["uid"]);
+			Logger::notice("The handle: " . $handle . " is not allowed to post to user " . $importer['uid']);
```
```suggestion
			Logger::notice('The handle: ' . $handle . ' is not allowed to post to user ' . $importer['uid']);
```

MrPetovan commented on src/Protocol/Diaspora.php:
```diff
@@ -916,8 +923,8 @@ private static function messageExists(int $uid, string $guid)
 		if (DBA::isResult($item)) {
-			Logger::notice("message ".$guid." already exists for user ".$uid);
-			return $item["id"];
+			Logger::notice("message " . $guid . " already exists for user " . $uid);
```
```suggestion
			Logger::notice('message ' . $guid . ' already exists for user ' . $uid);
```

MrPetovan commented on src/Protocol/Diaspora.php:
```diff
@@ -1016,21 +1022,21 @@ private static function storeByGuid(string $guid, string $server, bool $force)
+		$server = $serverparts['scheme'] . '://' . $serverparts['host'];
 
-		Logger::info("Trying to fetch item ".$guid." from ".$server);
+		Logger::info("Trying to fetch item " . $guid . " from " . $server);
```
```suggestion
		Logger::info('Trying to fetch item ' . $guid . ' from ' . $server);
```

MrPetovan commented on src/Protocol/Diaspora.php:
```diff
@@ -1016,21 +1022,21 @@ private static function storeByGuid(string $guid, string $server, bool $force)
 		}
 
-		Logger::info("Successfully fetched item ".$guid." from ".$server);
+		Logger::info("Successfully fetched item " . $guid . " from " . $server);
```
```suggestion
		Logger::info('Successfully fetched item ' . $guid . ' from ' . $server);
```

MrPetovan commented on src/Protocol/Diaspora.php:
```diff
@@ -1056,9 +1062,9 @@ public static function message(string $guid, string $server, int $level = 0)
+		$source_url = $server . '/fetch/post/' . urlencode($guid);
 
-		Logger::info("Fetch post from ".$source_url);
+		Logger::info("Fetch post from " . $source_url);
```
```suggestion
		Logger::info('Fetch post from ' . $source_url);
```

MrPetovan commented on src/Protocol/Diaspora.php:
```diff
@@ -1176,25 +1182,25 @@ private static function parentItem(int $uid, string $guid, string $author, array
 
 			if ($result) {
-				Logger::info("Fetched missing item ".$guid." - result: ".$result);
+				Logger::info("Fetched missing item " . $guid . " - result: " . $result);
```
```suggestion
				Logger::info('Fetched missing item ' . $guid . ' - result: ' . $result);
```

MrPetovan commented on src/Protocol/Diaspora.php:
```diff
@@ -1176,25 +1182,25 @@ private static function parentItem(int $uid, string $guid, string $author, array
 
 		if (!DBA::isResult($item)) {
-			Logger::notice("parent item not found: parent: ".$guid." - user: ".$uid);
+			Logger::notice("parent item not found: parent: " . $guid . " - user: " . $uid);
```
```suggestion
			Logger::notice('parent item not found: parent: ' . $guid . ' - user: ' . $uid);
```

MrPetovan commented on src/Protocol/Diaspora.php:
```diff
@@ -1176,25 +1182,25 @@ private static function parentItem(int $uid, string $guid, string $author, array
 			return false;
 		} else {
-			Logger::notice("parent item found: parent: ".$guid." - user: ".$uid);
+			Logger::notice("parent item found: parent: " . $guid . " - user: " . $uid);
```
```suggestion
			Logger::notice('parent item found: parent: ' . $guid . ' - user: ' . $uid);
```

MrPetovan commented on src/Protocol/Diaspora.php:
```diff
@@ -1316,23 +1325,23 @@ private static function plink(string $addr, string $guid, string $parent_guid =
+		$contact = self::contactByHandle($importer['uid'], $old_handle);
 		if (!$contact) {
-			Logger::notice("cannot find contact for sender: ".$old_handle." and user ".$importer["uid"]);
+			Logger::notice("cannot find contact for sender: " . $old_handle . " and user " . $importer['uid']);
```
```suggestion
			Logger::notice('cannot find contact for sender: ' . $old_handle . ' and user ' . $importer['uid']);
```

MrPetovan commented on src/Protocol/Diaspora.php:
```diff
@@ -1316,23 +1325,23 @@ private static function plink(string $addr, string $guid, string $parent_guid =
 		}
 
-		Logger::notice("Got migration for ".$old_handle.", to ".$new_handle." with user ".$importer["uid"]);
+		Logger::notice("Got migration for " . $old_handle . ", to " . $new_handle . " with user " . $importer['uid']);
```
```suggestion
		Logger::notice('Got migration for ' . $old_handle . ', to ' . $new_handle . ' with user ' . $importer['uid']);
```

MrPetovan commented on src/Protocol/Diaspora.php:
```diff
@@ -1343,15 +1352,21 @@ private static function receiveAccountMigration(array $importer, $data): bool
 		$data = Probe::uri($new_handle);
 		if ($data['network'] == Protocol::PHANTOM) {
-			Logger::notice('Account for '.$new_handle." couldn't be probed.");
+			Logger::notice("Account for " . $new_handle . " couldn't be probed.");
```
```suggestion
			Logger::notice('Account for ' . $new_handle . ' couldn't be probed.');
```

MrPetovan commented on src/Protocol/Diaspora.php:
```diff
@@ -1578,7 +1593,7 @@ private static function receiveComment(array $importer, string $sender, $data, s
 
 		if ($message_id) {
-			Logger::info("Stored comment ".$datarray["guid"]." with message id ".$message_id);
+			Logger::info("Stored comment " . $datarray['guid'] . " with message id " . $message_id);
```
```suggestion
			Logger::info('Stored comment ' . $datarray['guid'] . ' with message id ' . $message_id);
```

MrPetovan commented on src/Protocol/Diaspora.php:
```diff
@@ -1827,7 +1842,7 @@ private static function receiveLike(array $importer, string $sender, $data, int
 
 		if ($message_id) {
-			Logger::info("Stored like ".$datarray["guid"]." with message id ".$message_id);
+			Logger::info("Stored like " . $datarray['guid'] . " with message id " . $message_id);
```
```suggestion
			Logger::info('Stored like ' . $datarray['guid'] . ' with message id ' . $message_id);
```

MrPetovan commented on src/Protocol/Diaspora.php:
```diff
@@ -2132,7 +2148,7 @@ private static function receiveProfile(array $importer, $data): bool
 		Contact::update($fields, ['id' => $contact['id']]);
 
-		Logger::info("Profile of contact ".$contact["id"]." stored for user ".$importer["uid"]);
+		Logger::info("Profile of contact " . $contact['id'] . " stored for user " . $importer['uid']);
```
```suggestion
		Logger::info('Profile of contact ' . $contact['id'] . ' stored for user ' . $importer['uid']);
```

MrPetovan commented on src/Protocol/Diaspora.php:
```diff
@@ -2178,64 +2194,64 @@ private static function receiveContactRequest(array $importer, $data): bool
 		if ($contact) {
 			if ($following) {
-				Logger::info("Author ".$author." (Contact ".$contact["id"].") wants to follow us.");
+				Logger::info("Author " . $author . " (Contact " . $contact['id'] . ") wants to follow us.");
```
```suggestion
				Logger::info('Author ' . $author . ' (Contact ' . $contact['id'] . ') wants to follow us.');
```

MrPetovan commented on src/Protocol/Diaspora.php:
```diff
@@ -2178,64 +2194,64 @@ private static function receiveContactRequest(array $importer, $data): bool
+					$user = DBA::selectFirst('user', [], ['uid' => $importer['uid']]);
 					if (DBA::isResult($user)) {
-						Logger::info("Sending share message to author ".$author." - Contact: ".$contact["id"]." - User: ".$importer["uid"]);
+						Logger::info("Sending share message to author " . $author . " - Contact: " . $contact['id'] . " - User: " . $importer['uid']);
```
```suggestion
						Logger::info('Sending share message to author ' . $author . ' - Contact: ' . $contact['id'] . ' - User: ' . $importer['uid']);
```

MrPetovan commented on src/Protocol/Diaspora.php:
```diff
@@ -2178,64 +2194,64 @@ private static function receiveContactRequest(array $importer, $data): bool
 				return true;
 			} else {
-				Logger::info("Author ".$author." doesn't want to follow us anymore.");
+				Logger::info("Author " . $author . " doesn't want to follow us anymore.");
```
```suggestion
				Logger::info('Author ' . $author . ' doesn't want to follow us anymore.');
```

MrPetovan commented on src/Protocol/Diaspora.php:
```diff
@@ -2178,64 +2194,64 @@ private static function receiveContactRequest(array $importer, $data): bool
-		if (!$following && $sharing && in_array($importer["page-flags"], [User::PAGE_FLAGS_SOAPBOX, User::PAGE_FLAGS_NORMAL])) {
-			Logger::info("Author ".$author." wants to share with us - but doesn't want to listen. Request is ignored.");
+		if (!$following && $sharing && in_array($importer['page-flags'], [User::PAGE_FLAGS_SOAPBOX, User::PAGE_FLAGS_NORMAL])) {
+			Logger::info("Author " . $author . " wants to share with us - but doesn't want to listen. Request is ignored.");
```
```suggestion
			Logger::info('Author ' . $author . " wants to share with us - but doesn't want to listen. Request is ignored.");
```

MrPetovan commented on src/Protocol/Diaspora.php:
```diff
@@ -2178,64 +2194,64 @@ private static function receiveContactRequest(array $importer, $data): bool
 			return false;
 		} elseif (!$following && !$sharing) {
-			Logger::info("Author ".$author." doesn't want anything - and we don't know the author. Request is ignored.");
+			Logger::info("Author " . $author . " doesn't want anything - and we don't know the author. Request is ignored.");
```
```suggestion
			Logger::info('Author ' . $author . " doesn't want anything - and we don't know the author. Request is ignored.");
```

MrPetovan commented on src/Protocol/Diaspora.php:
```diff
@@ -2178,64 +2194,64 @@ private static function receiveContactRequest(array $importer, $data): bool
 			return false;
 		} elseif (!$following && $sharing) {
-			Logger::info("Author ".$author." wants to share with us.");
+			Logger::info("Author " . $author . " wants to share with us.");
```
```suggestion
			Logger::info('Author ' . $author . ' wants to share with us.');
```

MrPetovan commented on src/Protocol/Diaspora.php:
```diff
@@ -2178,64 +2194,64 @@ private static function receiveContactRequest(array $importer, $data): bool
+			Logger::info("Author " . $author . " wants to share with us.");
 		} elseif ($following && $sharing) {
-			Logger::info("Author ".$author." wants to have a bidirectional conection.");
+			Logger::info("Author " . $author . " wants to have a bidirectional conection.");
```
```suggestion
			Logger::info('Author ' . $author . ' wants to have a bidirectional connection.');
```

MrPetovan commented on src/Protocol/Diaspora.php:
```diff
@@ -2178,64 +2194,64 @@ private static function receiveContactRequest(array $importer, $data): bool
+			Logger::info("Author " . $author . " wants to have a bidirectional conection.");
 		} elseif ($following && !$sharing) {
-			Logger::info("Author ".$author." wants to listen to us.");
+			Logger::info("Author " . $author . " wants to listen to us.");
```
```suggestion
			Logger::info('Author ' . $author . ' wants to listen to us.');
```

MrPetovan commented on src/Protocol/Diaspora.php:
```diff
@@ -2178,64 +2194,64 @@ private static function receiveContactRequest(array $importer, $data): bool
-		if (!$ret || ($ret["network"] != Protocol::DIASPORA)) {
-			Logger::notice("Cannot resolve diaspora handle ".$author." for ".$recipient);
+		if (!$ret || ($ret['network'] != Protocol::DIASPORA)) {
+			Logger::notice("Cannot resolve diaspora handle " . $author . " for ".$recipient);
```
```suggestion
			Logger::notice('Cannot resolve diaspora handle ' . $author . ' for ' . $recipient);
```

MrPetovan commented on src/Protocol/Diaspora.php:
```diff
@@ -2292,17 +2308,17 @@ public static function originalItem(string $guid, string $orig_author)
 
 		if (DBA::isResult($item)) {
-			Logger::notice("reshared message ".$guid." already exists on system.");
+			Logger::notice("reshared message " . $guid . " already exists on system.");
```
```suggestion
			Logger::notice('Reshared message ' . $guid . ' already exists on system.');
```

MrPetovan commented on src/Protocol/Diaspora.php:
```diff
@@ -2316,13 +2332,13 @@ public static function originalItem(string $guid, string $orig_author)
-			$server = "https://".substr($orig_author, strpos($orig_author, "@") + 1);
-			Logger::notice("1st try: reshared message ".$guid." will be fetched via SSL from the server ".$server);
+			$server = 'https://' . substr($orig_author, strpos($orig_author, '@') + 1);
+			Logger::notice("1st try: reshared message " . $guid . " will be fetched via SSL from the server " . $server);
```
```suggestion
			Logger::notice('1st try: reshared message ' . $guid . ' will be fetched via SSL from the server ' . $server);
```

MrPetovan commented on src/Protocol/Diaspora.php:
```diff
@@ -2316,13 +2332,13 @@ public static function originalItem(string $guid, string $orig_author)
-				$server = "http://".substr($orig_author, strpos($orig_author, "@") + 1);
-				Logger::notice("2nd try: reshared message ".$guid." will be fetched without SSL from the server ".$server);
+				$server = 'http://' . substr($orig_author, strpos($orig_author, '@') + 1);
+				Logger::notice("2nd try: reshared message " . $guid . " will be fetched without SSL from the server " . $server);
```
```suggestion
				Logger::notice('2nd try: reshared message ' . $guid . ' will be fetched without SSL from the server ' . $server);
```

MrPetovan commented on src/Protocol/Diaspora.php:
```diff
@@ -2509,13 +2525,13 @@ private static function receiveReshare(array $importer, $data, string $xml, int
 
 		if ($message_id) {
-			Logger::info("Stored reshare ".$datarray["guid"]." with message id ".$message_id);
+			Logger::info("Stored reshare " . $datarray['guid'] . " with message id " . $message_id);
```
```suggestion
			Logger::info('Stored reshare ' . $datarray['guid'] . ' with message id ' . $message_id);
```

MrPetovan commented on src/Protocol/Diaspora.php:
```diff
@@ -2545,12 +2561,12 @@ private static function itemRetraction(array $importer, array $contact, $data):
 		$person = FContact::getByURL($author);
 		if (!is_array($person)) {
-			Logger::notice("unable to find author detail for ".$author);
+			Logger::notice("unable to find author detail for " . $author);
```
```suggestion
			Logger::notice('Unable to find author detail for ' . $author);
```

MrPetovan commented on src/Protocol/Diaspora.php:
```diff
@@ -2565,7 +2581,7 @@ private static function itemRetraction(array $importer, array $contact, $data):
 		$r = Post::select($fields, $condition);
 		if (!DBA::isResult($r)) {
-			Logger::notice("Target guid ".$target_guid." was not found on this system for user ".$importer['uid'].".");
+			Logger::notice("Target guid " . $target_guid . " was not found on this system for user " . $importer['uid'] . ".");
```
```suggestion
			Logger::notice('Target guid ' . $target_guid . ' was not found on this system for user ' . $importer['uid']);
```

MrPetovan commented on src/Protocol/Diaspora.php:
```diff
@@ -2579,14 +2595,14 @@ private static function itemRetraction(array $importer, array $contact, $data):
-			if (!Strings::compareLink($parent["author-link"], $contact["url"]) && !Strings::compareLink($item["author-link"], $contact["url"])) {
-				Logger::info("Thread author ".$parent["author-link"]." and item author ".$item["author-link"]." don't fit to expected contact ".$contact["url"]);
+			if (!Strings::compareLink($parent['author-link'], $contact['url']) && !Strings::compareLink($item['author-link'], $contact['url'])) {
+				Logger::info("Thread author " . $parent['author-link'] . " and item author " . $item['author-link'] . " don't fit to expected contact " . $contact['url']);
```
```suggestion
				Logger::info('Thread author ' . $parent['author-link'] . ' and item author ' . $item['author-link'] . " don't fit to expected contact " . $contact['url']);
```

MrPetovan commented on src/Protocol/Diaspora.php:
```diff
@@ -2579,14 +2595,14 @@ private static function itemRetraction(array $importer, array $contact, $data):
 			Item::markForDeletion(['id' => $item['id']]);
 
-			Logger::info("Deleted target ".$target_guid." (".$item["id"].") from user ".$item["uid"]." parent: ".$item['parent']);
+			Logger::info("Deleted target " . $target_guid . " (" . $item['id'] . ") from user " . $item['uid'] . " parent: " . $item['parent']);
```
```suggestion
			Logger::info('Deleted target ' . $target_guid . ' (' . $item['id'] . ') from user ' . $item['uid'] . ' parent: ' . $item['parent']);
```

MrPetovan commented on src/Protocol/Diaspora.php:
```diff
@@ -2609,33 +2625,33 @@ private static function receiveRetraction(array $importer, string $sender, $data
-			Logger::notice("cannot find contact for sender: ".$sender." and user ".$importer["uid"]);
+		$contact = self::contactByHandle($importer['uid'], $sender);
+		if (!$contact && (in_array($target_type, ['Contact', 'Person']))) {
+			Logger::notice("cannot find contact for sender: " . $sender . " and user " . $importer['uid']);
```
```suggestion
			Logger::notice('Cannot find contact for sender: ' . $sender . ' and user ' . $importer['uid']);
```

MrPetovan commented on src/Protocol/Diaspora.php:
```diff
@@ -2609,33 +2625,33 @@ private static function receiveRetraction(array $importer, string $sender, $data
 		}
 
-		Logger::info("Got retraction for ".$target_type.", sender ".$sender." and user ".$importer["uid"]);
+		Logger::info("Got retraction for " . $target_type . ", sender " . $sender . " and user " . $importer['uid']);
```
```suggestion
		Logger::info('Got retraction for ' . $target_type . ', sender ' . $sender . ' and user ' . $importer['uid']);
```

MrPetovan commented on src/Protocol/Diaspora.php:
```diff
@@ -2609,33 +2625,33 @@ private static function receiveRetraction(array $importer, string $sender, $data
 
 			default:
-				Logger::notice("Unknown target type ".$target_type);
+				Logger::notice("Unknown target type " . $target_type);
```
```suggestion
				Logger::notice('Unknown target type ' . $target_type);
```

MrPetovan commented on src/Protocol/Diaspora.php:
```diff
@@ -2833,7 +2848,7 @@ private static function receiveStatusMessage(array $importer, SimpleXMLElement $
 
 		if ($message_id) {
-			Logger::info("Stored item ".$datarray["guid"]." with message id ".$message_id);
+			Logger::info("Stored item " . $datarray['guid'] . " with message id " . $message_id);
```
```suggestion
			Logger::info('Stored item ' . $datarray['guid'] . ' with message id ' . $message_id);
```

MrPetovan commented on src/Protocol/Diaspora.php:
```diff
@@ -3029,32 +3050,32 @@ private static function transmit(array $owner, array $contact, string $envelope,
 
 		if (!$dest_url) {
-			Logger::notice("no url for contact: ".$contact["id"]." batch mode =".$public_batch);
+			Logger::notice("no url for contact: " . $contact['id'] . " batch mode =" . $public_batch);
```
```suggestion
			Logger::notice('No url for contact: ' . $contact['id'] . ' batch mode = ' . $public_batch);
```

MrPetovan commented on src/Protocol/Diaspora.php:
```diff
@@ -3029,32 +3050,32 @@ private static function transmit(array $owner, array $contact, string $envelope,
 		}
 
-		Logger::notice("transmit: ".$logid."-".$guid." ".$dest_url);
+		Logger::notice("transmit: " . $logid . "-" . $guid . " " . $dest_url);
```
```suggestion
		Logger::notice('transmit: ' . $logid . '-' . $guid . ' ' . $dest_url);
```

MrPetovan commented on src/Protocol/Diaspora.php:
```diff
@@ -3029,32 +3050,32 @@ private static function transmit(array $owner, array $contact, string $envelope,
 		}
 
-		Logger::notice("transmit: ".$logid."-".$guid." to ".$dest_url." returns: ".$return_code);
+		Logger::notice("transmit: " . $logid . "-" . $guid . " to " . $dest_url . " returns: " . $return_code);
```
```suggestion
		Logger::notice('transmit: ' . $logid . '-' . $guid . ' to ' . $dest_url . ' returns: ' . $return_code);
```

MrPetovan commented on src/Protocol/Diaspora.php:
```diff
@@ -3161,17 +3182,19 @@ private static function sendParticipation(array $contact, array $item): int
+		];
 
-		Logger::info("Send participation for ".$item["guid"]." by ".$author);
+		Logger::info("Send participation for " . $item['guid'] . " by " . $author);
```
```suggestion
		Logger::info('Send participation for ' . $item['guid'] . ' by ' . $author);
```

MrPetovan commented on src/Protocol/Diaspora.php:
```diff
@@ -3730,41 +3770,41 @@ public static function sendFollowup(array $item, array $owner, array $contact, b
 		}
 
-		Logger::info("Got relayable data ".$type." for item ".$item["guid"]." (".$item["id"].")");
+		Logger::info("Got relayable data " . $type . " for item " . $item['guid'] . " (" . $item['id'] . ")");
```
```suggestion
		Logger::info('Got relayable data ' . $type . ' for item ' . $item['guid'] . ' (' . $item['id'] . ')');
```

MrPetovan commented on src/Protocol/Diaspora.php:
```diff
@@ -3730,41 +3770,41 @@ public static function sendFollowup(array $item, array $owner, array $contact, b
 			}
 		} else {
-			Logger::info("Signature text for item ".$item["guid"]." (".$item["id"].") couldn't be extracted: ".$item['signed_text']);
+			Logger::info("Signature text for item " . $item["guid"] . " (" . $item["id"] . ") couldn't be extracted: " . $item['signed_text']);
```
```suggestion
			Logger::info('Signature text for item ' . $item['guid'] . ' (' . $item['id'] . ") couldn't be extracted: " . $item['signed_text']);
```

MrPetovan commented on src/Protocol/Diaspora.php:
```diff
@@ -3817,40 +3859,40 @@ public static function sendMail(array $item, array $owner, array $contact): int
-		$cnv = DBA::selectFirst('conv', [], ['id' => $item["convid"], 'uid' => $item["uid"]]);
+		$cnv = DBA::selectFirst('conv', [], ['id' => $item['convid'], 'uid' => $item['uid']]);
 		if (!DBA::isResult($cnv)) {
 			Logger::notice("conversation not found.");
```
```suggestion
			Logger::notice('Conversation not found.');
```

nupplaphil commented:
I'm sorry that I don't have enough time to contribute more than these few sentences, but when we already refactor the logging points, we should at least extract the parameters from the message, like I commented at one place :-)

Quix0r committed:
Continued:
- prevents a "Return value of Friendica\Core\Worker::workerProcess() must be of the type array, bool returned"

Quix0r commented:
This ticket has nothing to do with rewriting log messages, it wasn't even about changing double-quotes to single as requested by @MrPetovan . This ticket was only about type-hints and now doesn't get accepted because more rewrites (off-topic rewrites) are being requested. Let's really move that whole thing into another PR/branch and not make this lengthy ticket more and more longer.

Quix0r commented:
This ticket already covers 37 commits and 50 files being touched. I guess staying on topic is more healthier than requesting off-topic stuff that doesn't belong in here.

Quix0r commented:
Okay, let's move the rewrite of log messages to another branch/PR. Thank you @MrPetovan and @nupplaphil . :-)
